azurerm/resource_arm_app_service.go

```diff
@@ -656,12 +668,28 @@ func flattenAppServiceAppSettings(input map[string]*string) map[string]string {
 }
 
 func expandAzureRmAppServiceIdentity(d *schema.ResourceData) *web.ManagedServiceIdentity {
-	identities := d.Get("identity").([]interface{})
+	v := d.Get("identity")
+	identities := v.([]interface{})
+	if len(identities) == 0 {
+		return nil
+	}
 	identity := identities[0].(map[string]interface{})
-	identityType := identity["type"].(string)
-	return &web.ManagedServiceIdentity{
-		Type: web.ManagedServiceIdentityType(identityType),
+	identityType := web.ManagedServiceIdentityType(identity["type"].(string))
+
+	identityIds := make(map[string]*web.ManagedServiceIdentityUserAssignedIdentitiesValue)
+	for _, id := range identity["identity_ids"].([]interface{}) {
```

Comment:
I think your going to need a nil check here:
```
Test ended in panic.

------- Stdout: -------
=== RUN   TestAccAzureRMAppServiceSlot_enableManageServiceIdentity
=== PAUSE TestAccAzureRMAppServiceSlot_enableManageServiceIdentity
=== CONT  TestAccAzureRMAppServiceSlot_enableManageServiceIdentity

------- Stderr: -------
panic: interface conversion: interface {} is nil, not []interface {}

goroutine 326 [running]:
github.com/terraform-providers/terraform-provider-azurerm/azurerm.expandAzureRmAppServiceIdentity(0xc000144700, 0x3755fc0)
	/opt/teamcity-agent/work/458e5e4800bd94f6/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_app_service.go:680 +0x37a
github.com/terraform-providers/terraform-provider-azurerm/azurerm.resourceArmAppServiceSlotCreate(0xc000144700, 0x3684c80, 0xc00136e000, 0x2, 0x6ca1720)
	/opt/teamcity-agent/work/458e5e4800bd94f6/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_app_service_slot.go:210 +0x1048
github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Apply(0xc000499f00, 0xc000d5e460, 0xc000fff8a0, 0x3684c80, 0xc00136e000, 0x32b9301, 0xc000a49078, 0xc001129230)
	/opt/teamcity-agent/work/458e5e4800bd94f6/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:286 +0x363
github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Apply(0xc0004e5080, 0xc000909a00, 0xc000d5e460, 0xc000fff8a0, 0xc000a48f68, 0xc000c200e0, 0x32bc0c0)
	/opt/teamcity-agent/work/458e5e4800bd94f6/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:285 +0x9c
github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc0009dd910, 0x3e0ecc0, 0xc000ace4b0, 0xc00072b6e0, 0xc0009dd910, 0xc000ace420, 0x331e4c0)
	/opt/teamcity-agent/work/458e5e4800bd94f6/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/plugin/grpc_provider.go:842 +0x87a
github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x35ee460, 0xc0009dd910, 0x3e0ecc0, 0xc000ace4b0, 0xc000d5e280, 0x0, 0x0, 0x0, 0xc000520500, 0x4a2)
	/opt/teamcity-agent/work/458e5e4800bd94f6/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/internal/tfplugin5/tfplugin5.pb.go:3019 +0x23e
github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc.(*Server).processUnaryRPC(0xc00127ed80, 0x3e2c9a0, 0xc00127f800, 0xc000db4000, 0xc000f17800, 0x6c694e0, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/458e5e4800bd94f6/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:966 +0x4a2
github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc.(*Server).handleStream(0xc00127ed80, 0x3e2c9a0, 0xc00127f800, 0xc000db4000, 0x0)
	/opt/teamcity-agent/work/458e5e4800bd94f6/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1245 +0xd61
github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc000b78560, 0xc00127ed80, 0x3e2c9a0, 0xc00127f800, 0xc000db4000)
	/opt/teamcity-agent/work/458e5e4800bd94f6/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:685 +0x9f
created by github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
	/opt/teamcity-agent/work/458e5e4800bd94f6/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:683 +0xa1
```

Reply:
Sorry, dont know how I missed the `azurerm_app_service_slot` resource here.